### PR TITLE
Fix templates for NodeFeatureRule with MatchAny

### LINKED ISSUE #864: Templates don't work on NodeFeatureRule with MatchAny only
**What happened**:
I've stumbled into it while sketching a PR for adding annotations with nfd, so I'm opening this issue just for tracking purposes.

**How to reproduce it (as minimally and precisely as possible)**:
Run tests from submitted PR without the fix in submitted PR.

### PR DESCRIPTION
**Pull request by stek29**
When `MatchAny` is matched, template existence is checked with `r.labelsTemplate == nil`, and if it's nil - templating is skipped.

This is wrong because:
- there are `vars` templates too, not only `labels` templates
- `labelsTemplate` is initialized from `LabelsTemplate` when `executeLabelsTemplate` is called first time, so it would always be nil unless Rule has `MatchFeatures` in addition to `MatchAny`, and was executed already.

`LabelsTemplate` should be checked instead of `labelsTemplate` (same for vars) to see if there's template – but `executeLabelsTemplate` already does that.

So the check should be removed completely.

I've also added a test case which catches this issue.

Fixes #864.

### CODE REVIEW

File: pkg/apis/nfd/v1alpha1/rule.go [hunk 1]
```diff
@@ -50,11 +50,6 @@ func (r *Rule) Execute(features feature.Features) (RuleOutput, error) {
 				matched = true
 				utils.KlogDump(4, "matches for matchAny "+r.Name, "  ", matches)
 
-				if r.labelsTemplate == nil {
-					// No templating so we stop here (further matches would just
-					// produce the same labels)
-					break
-				}
```
Comment (marquiz) [inline pkg/apis/nfd/v1alpha1/rule.go:57]: It is still useful to check if the templates have been defined. If there would be 10 items under `MatchAny` and the first matches (and no templates are specified) there is no use to evaluate the 9 remaining items.

My suggestion to change/fix this instead

```
                                if r.LabelsTemplate == "" && r.VarsTemplate == "" {
                                        // No templating so we stop here (further matches would just
                                        // produce the same labels / vars)
                                        break
                                }
```

**Review by marquiz — COMMENTED** [review]
Wov, thanks for the patch @stek29! This is indeed broken. And great that you updated the unit tests, too 👍

I had one suggestion, though.

/ok-to-test

**Comment by marquiz** [comment]
...and seems that somebody's really using the templating functionality 😊 

/cherry-pick release-0.11

**Comment by ArangoGutierrez** [comment]
PING @stek29

File: pkg/apis/nfd/v1alpha1/rule.go (hunk 1 above)
Comment (stek29) [reply pkg/apis/nfd/v1alpha1/rule.go:57]: sure, will do

**Review by marquiz — APPROVED** [review]
Looks good to me now 👌

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**